sklearn/feature_selection/multivariate_filtering.py

```diff
@@ -0,0 +1,154 @@
+                Relevance of all the features
+    redundancy : array, shape=[n_features]
+                 Redundancy of all the features
+    rule : string, default='diff'
+           Rule to combine relevance and redundancy, either
+           'diff' - difference between the two
+           'prod' - product between the two
+    X : array, shape=[n_samples, n_features]
+        Input dataset, must be either integer or categorical
+    y : array, shape=[n_samples]
+        Label vector, must be either integer or categorical
+
+    References
+    ----------
+    .. [1] H. Peng, F. Long, and C. Ding, "Feature selection based on mutual
+       information: criteria of max-dependency, max-relevance, and
+       min-redundancy", IEEE Transactions on Pattern Analysis and Machine
+       Intelligence, Vol. 27, No. 8, pp.1226-1238, 2005.
+    """
+    def __init__(self, k=2, rule='diff'):
+        """
+        Parameters
+        ----------
+        k : int, default=2
+            Number of features to select
+        rule : string, default='diff'
+               Rule to combine relevance and redundancy, either
+               'diff' - difference between the two
+               'prod' - product between the two
+        """
+        self.k = k
+        self.rule = rule
+
+    def fit(self, X, y):
+        """
+        Parameters
+        ----------
+        X : array, shape=[n_samples, n_features]
+            Input dataset, must be either integer or categorical
+        y : array, shape=[n_samples]
+            Label vector, must be either integer or categorical
+        """
+        X = array2d(X)
+
+        self.X = X
+        self.y = y
+        self.mask, self.score = self._compute_mRMR(X, y)
+        return self
+
+    def _get_support_mask(self):
+        """
+        Returns
+        -------
+        support : array, dype=bool, shape=[n_features]
+                  Boolean mask with True the selected features
+        """
+
+        support = np.zeros(self.n_features, dtype=bool)
+        support[self.mask] = True
+        return support
+
+    def _compute_mRMR(self, X, y):
+        """
+        Parameters
+        ----------
+        X : array, shape=[n_samples, n_features]
+            Input dataset, must be either integer or categorical
+        y : array, shape=[n_samples]
+            Label vector, must be either integer or categorical
+
+        Returns
+        -------
+        mask : list, len=selected_features
+               Integer list of the features ordered by maximal relevance and
+               minimal redundancy
+        score : list, len=selected_features
+                mRMR score associated to each entry in mask
+        """
+        M = X.shape[1]  # Number of features
+
+        # Computation of relevance and redundancy
+        relevance = np.zeros(M)
+        redundancy = np.zeros([M, M])
+        for m1 in range(0, M):
+            relevance[m1] = mutual_info_score(X[:, m1], y)
```

Comment:
It would be nice if we could reimplement mutual_info to support calculation over sparse feature spaces (in a CSC matrix).